index.js

```diff
@@ -167,6 +167,27 @@ export default class Supercluster {
         return expansionZoom;
     }
 
+    getPointUnclusterZoom(point) {
+        const [lng, lat] = point;
+        const pointX = fround(lngX(lng));
+        const pointY = fround(latY(lat));
+
+        let expansionZoom = 0;
+        while (expansionZoom <= this.options.maxZoom) {
+            const tree = this.trees[expansionZoom];
+
+            const unclustered = tree.points.some(
+                treePoint => !treePoint.id && treePoint.x === pointX && treePoint.y === pointY
+            );
```

Comment:
This essentially loops through all clusters across all zooms until it reaches the unclustered point, in many instances iterating over nearly the whole dataset. I think we could implement this much more efficiently, utilizing the R-tree indices — e.g. only look at clusters within `radius` of the point.

Also, what if the point does have `id`? This filter doesn't look right...

Reply:
Thank you for your review and advises!

I improved the code and used the `within` function of KDBush to filter all points that don't have the same coordinate as the provided point, by setting radius to 0.

I mistakenly assumed that only clusters have an id, so I wanted to filter them first. The value of the `parentId` should be better suited for this.

Additionally the `expansionZoom`  now starts at the `minZoom` level, instead of 0, to avoid getting undefined `trees` entries.

If you have some other advises for improvements, please let me know.